regtest/bitcoind.js

```diff
@@ -0,0 +1,461 @@
+                      finished();
+                    });
+                  }
+                ], next);
+              }, function(err) {
+                if (err) {
+                  throw err;
+                }
+                done();
+              });
+            });
+          });
+        });
+      });
+    });
+  });
+
+  after(function(done) {
+    this.timeout(20000);
+    bitcoind.node.stopping = true;
+    bitcoind.stop(function(err, result) {
+      done();
+    });
+  });
+
+  describe('get blocks by hash', function() {
+
+    [0,1,2,3,5,6,7,8,9].forEach(function(i) {
+      it('generated block ' + i, function(done) {
+        bitcoind.getBlock(blockHashes[i], function(err, block) {
+          if (err) {
+            throw err;
+          }
+          should.exist(block);
+          block.hash.should.equal(blockHashes[i]);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('get blocks as buffers', function() {
+    [0,1,2,3,5,6,7,8,9].forEach(function(i) {
+      it('generated block ' + i, function(done) {
+        bitcoind.getRawBlock(blockHashes[i], function(err, block) {
+          if (err) {
+            throw err;
+          }
+          should.exist(block);
+          (block instanceof Buffer).should.equal(true);
+          done();
+        });
+      });
+    });
+  });
+
+  describe('get errors as error instances', function() {
+    it('will wrap an rpc into a javascript error', function(done) {
+      bitcoind.client.getBlock(1000000000, function(err, response) {
+        var error = bitcoind._wrapRPCError(err);
+        (error instanceof Error).should.equal(true);
+        error.message.should.equal(err.message);
+        error.code.should.equal(err.code);
+        should.exist(error.stack);
+        done();
+      });
+    });
+  });
+
+  describe('get blocks by height', function() {
+
+    [0,1,2,3,4,5,6,7,8,9].forEach(function(i) {
+      it('generated block ' + i, function(done) {
+        // add the genesis block
+        var height = i + 1;
+        bitcoind.getBlock(i + 1, function(err, block) {
+          if (err) {
+            throw err;
+          }
+          should.exist(block);
+          block.hash.should.equal(blockHashes[i]);
+          done();
+        });
+      });
+    });
+
+    it('will get error with number greater than tip', function(done) {
+      bitcoind.getBlock(1000000000, function(err, response) {
+        should.exist(err);
+        done();
+      });
+    });
+
+  });
+
+  describe('get transactions by hash', function() {
+    [0,1,2,3,4,5,6,7,8,9].forEach(function(i) {
+      it('for tx ' + i, function(done) {
+        var txhex = transactionData[i];
+        var tx = new bitcore.Transaction();
+        tx.fromString(txhex);
+        bitcoind.getTransaction(tx.hash, function(err, response) {
+          if (err) {
+            throw err;
+          }
+          assert(response.toString('hex') === txhex, 'incorrect tx data result');
```

Comment:
We should check that the result is an instanceof a Buffer here.

If the result was a string instead of a buffer, this would call .toString('hex') on the string, which would still yield the equivalent of txhex.